src/core/layout/qgslayoutitemlegend.cpp

```diff
@@ -1484,6 +1493,85 @@ bool QgsLayoutItemLegend::isRefreshing() const
 }
 
 
+void QgsLayoutItemLegend::syncLayersWithUpdatedCanvasMinimumMaximum()
+{
+
+  if ( mCustomLayerTree )
+    return;
+
+  if ( !mMap )
+    return;
+
+  QgsLegendModel *model = this->model();
+  if ( !model )
+    return;
+
+  QgsLayerTreeGroup *root = model->rootGroup();
+  if ( !root )
+    return;
+
+  const QList<QgsLayerTreeLayer *> layers = root->findLayers();
```

Comment:
Can you move all this logic to a function in QgsLayerTreeUtils ? I think it potentially has use outside of the layout items.